Sources/Shared/Requests/Request+PublishToSocial.swift

```diff
@@ -36,72 +36,73 @@ public extension Request where ModelType == PublishJob {
+    ///   - videoURI: The URI for the video to be published.
+    static func publishPosts(_ posts: SocialMediaPosts, for videoURI: String) -> Request {
+        let publishToSocialPath = Request.uri(for: videoURI)
         var parameters = [String: Any]()
 
         posts.facebook.map {
-            var post = [String: Any]()
-            post[String.Key.title] = $0.title
-            post[String.Key.description] = $0.description
-            post[String.Key.destination] = $0.destination
-            post[String.Key.categoryID] = $0.categoryID
-            post[String.Key.allowEmbedding] = $0.allowEmbedding
-            post[String.Key.shouldAppearOnNewsFeed] = $0.shouldAppearOnNewsFeed
-            post[String.Key.isSecretVideo] = $0.isSecretVideo
-            post[String.Key.allowSocialActions] = $0.allowSocialActions
-            parameters[String.Key.facebook] = post
+            var post: [String: Any] = [
+                .title: $0.title,
+                .description: $0.description,
+                .destination: $0.destination,
+                .allowEmbedding: $0.allowEmbedding,
+                .shouldAppearOnNewsFeed: $0.shouldAppearOnNewsFeed,
+                .isSecretVideo: $0.isSecretVideo,
+                .allowSocialActions: $0.allowSocialActions
+            ]
+            
+            $0.categoryID.map { (categoryID) in post[.categoryID] = categoryID }
+            parameters[.facebook] = post
         }
 
         posts.linkedIn.map {
-            var post = [String: Any]()
-            post[String.Key.pageID] = $0.pageID
-            post[String.Key.title] = $0.title
-            post[String.Key.description] = $0.description
-            parameters[String.Key.linkedin] = post
+            parameters[.linkedin] = [
+                String.pageID: $0.pageID,
+                String.title: $0.title,
+                String.description: $0.description,
+            ]
         }
 
         posts.twitter.map {
-            var post = [String: String]()
-            post[String.Key.tweet] = $0.tweet
-            parameters[String.Key.twitter] = post
+            parameters[.twitter] = [
+                String.tweet: $0.tweet
+            ]
         }
 
         posts.youTube.map {
-            var post = [String: Any]()
-            post[String.Key.title] = $0.title
-            post[String.Key.description] = $0.description
-            post[String.Key.privacy] = $0.privacy.rawValue
-            post[String.Key.categoryID] = $0.categoryID
-            $0.tags.map { (tags) in  post[String.Key.tags] = tags }
-            parameters[String.Key.youtube] = post
+            var post: [String: Any] = [
+                .title: $0.title,
+                .privacy: $0.privacy.rawValue,
+                .categoryID: $0.categoryID
+            ]
+
+            $0.description.map { (description) in post[.description] = description }
+            $0.tags.map { (tags) in post[.tags] = tags }
```

Comment:
Since `tags` is an optional array, how does this map? Does it pass the entire array or does it only take the last value?

Reply:
Good question, this will assign the entire array. This should be validated in `Request+PublishToSocialTests.swift:158`:

```Swift
let socialMediaPosts = SocialMediaPosts(youTube: youTubePost)
let request = Request.publishPosts(socialMediaPosts, for: "/videos/34567")
let parameters = try XCTUnwrap(request.parameters as? [String: Any])
let youtubeParameters = try XCTUnwrap(parameters["youtube"] as? [String: AnyHashable])
…
XCTAssertEqual(youtubeParameters["tags"], ["test", "tags"])
```